solc/main.cpp

```diff
@@ -29,6 +35,15 @@ using namespace std;
 int main(int argc, char** argv)
 {
 	dev::solidity::CommandLineInterface cli;
+
+	if (isatty(fileno(stdin)) && argc == 1)
+	{
+		argc = 2;
+		argv++;
+		argv[0] = (char*)("solc");
+		argv[1] = (char*)("--help");
+		argv[2] = NULL;
```

Comment:
Thats why we check that argc is 1...that way we know argv[2] is invalid and fine to set to null.